src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -996,6 +1041,38 @@ void Esp32WiFiManager::enable_esp_wifi_logging()
 #endif // WIFI_LOG_SUBMODULE_ALL
 }
 
+void Esp32WiFiManager::start_ssid_scan(Notifiable *n)
+{
+    OSMutexLock l(&ssidScanResultsLock_);
+    ssidScanResults_.clear();
+    ssidCompleteNotifiable_ = n;
```

Comment:
you need to check that the notifiable is nullptr before assigning. Losing notifiables usually causes a lockup.

Reply:
something like (with proper formatting) this?
```
auto old = std::swap(ssidCompleteNotifiable_, n);
if(old) old->notify();
```

Reply:
also should the ssidCompleteNotifiable_ be set to nullptr *AFTER* the scan completes and notify() has been called?

Reply:
opted for setting to null.